pkg/server/registry/apigroups/acorn/projects/validator.go

```diff
@@ -58,43 +58,57 @@ func (v *Validator) ValidateUpdate(ctx context.Context, new, old runtime.Object)
 	}
 
 	if len(removedRegions) > 0 {
-		return v.ensureNoObjectsExistInRegions(ctx, newProject.Name, removedRegions, &apiv1.AppList{}, &apiv1.VolumeList{})
+		return v.ensureNoObjectsExistInRegions(ctx, newProject.Name, newProject.Spec.SupportedRegions, removedRegions, &apiv1.AppList{}, &apiv1.VolumeList{})
 	}
 
 	return nil
 }
 
-func (v *Validator) ensureNoObjectsExistInRegions(ctx context.Context, namespace string, regions []string, objList ...kclient.ObjectList) field.ErrorList {
+func (v *Validator) ensureNoObjectsExistInRegions(ctx context.Context, namespace string, regions, removedRegions []string, objList ...kclient.ObjectList) field.ErrorList {
 	var result field.ErrorList
 	for _, obj := range objList {
-		objectsInRemoveRegions := make([]string, 0)
+		var inRemovedRegion []string
 		if err := v.Client.List(ctx, obj, kclient.InNamespace(namespace)); err != nil {
 			return field.ErrorList{field.InternalError(field.NewPath("spec", "supportedRegions"), err)}
 		}
 
 		if err := meta.EachListItem(obj, func(object runtime.Object) error {
 			regionObject, ok := object.(regionNamer)
-			if ok && slices.Contains(regions, regionObject.GetRegion()) {
-				objectsInRemoveRegions = append(objectsInRemoveRegions, regionObject.GetName())
+			if ok && slices.Contains(removedRegions, regionObject.GetRegion()) {
+				inRemovedRegion = append(inRemovedRegion, regionObject.GetName())
 			}
 			return nil
 		}); err != nil {
 			return field.ErrorList{field.InternalError(field.NewPath("spec", "supportedRegions"), err)}
 		}
 
-		if len(objectsInRemoveRegions) > 0 {
+		if len(inRemovedRegion) > 0 {
 			result = append(result, field.Invalid(
 				field.NewPath("spec", "supportedRegions"),
 				regions,
```

Comment:
`field.Invalid` takes the _new_ field value as an argument here; before we were feeding it the removed regions.